pennylane/measurements/mid_measure.py

```diff
@@ -99,68 +105,131 @@ class MeasurementValue(Generic[T]):
     Measurements on a single qubit in the computational basis are assumed.
 
     Args:
-        measurement_id (str): The id of the measurement that this object depends on.
-        zero_case (float): the first measurement outcome value
-        one_case (float): the second measurement outcome value
+        measurement_ids (list of str): The id of the measurement that this object depends on.
+        fn (callable): A transformation applied to the measurements.
     """
 
-    __slots__ = ("_depends_on", "_zero_case", "_one_case", "_control_value")
-
-    def __init__(
-        self,
-        measurement_id: str,
-        zero_case: float = 0,
-        one_case: float = 1,
-    ):
-        self._depends_on = measurement_id
-        self._zero_case = zero_case
-        self._one_case = one_case
-        self._control_value = one_case  # By default, control on the one case
+    def __init__(self, measurement_ids, fn):
+        self.measurement_ids = measurement_ids
+        self.fn = fn
 
-    @property
-    def branches(self):
-        """A dictionary representing all the possible outcomes of the MeasurementValue."""
-        branch_dict = {}
-        branch_dict[(0,)] = self._zero_case
-        branch_dict[(1,)] = self._one_case
-        return branch_dict
+    def _items(self):
```

Comment:
hmm good question, I guess I was thinking it's not a necessary feature (therefore reducing testing surface area). and then I changed the output of the function to return a list rather than a dict, so it's an interface change.

Reply:
good point, I added the `branches` api back in